File: tests/indexing/parsing/lang/fixtures/rust/Formatting.rs
pub fn within<'a, T>(
    items: &'a [T],
    lower: &T,
    upper: &T,
) -> Vec<&'a T>
where
    T: PartialOrd,
{
    let mut kept = Vec::new();
    for item in items {
        if item >= lower
            && item <= upper
        {
            kept.push(item);
        }
    }
    kept
}

pub fn describe(kind: &Kind) -> &'static str {
    match kind {
        Kind::Alpha
        | Kind::Beta
        | Kind::Gamma => "early",
        Kind::Delta if kind.is_late() => {
            "late delta"
        }
        _ => "other",
    }
}

File: tests/indexing/parsing/lang/fixtures/swift/Formatting.swift
func within(
    values: [Int],
    lower: Int,
    upper: Int
) -> [Int] {
    var kept: [Int] = []
    for value in values {
        if value >= lower
            && value <= upper {
            kept.append(value)
        }
    }
    return kept
}

func categorize(code: Int) -> String {
    switch code {
    case 1,
        2,
        3:
        let mapped = code * 10
        return "low \(mapped)"
    default:
        return "high"
    }
}
